### PR TITLE
Register DebugAdapterExecutableFactory

### PR DESCRIPTION
**Pull request by JoeRobich**
Fixes https://github.com/OmniSharp/omnisharp-vscode/issues/3512

- Removes the "csharp.coreclrAdapterExecutableCommand" & "csharp.clrAdapterExecutableCommand"
- Registers a `DebugAdapterExecutableFactory` with the logic from the `getAdapterExecutionCommand` method.

cc: @gregg-miskelly, @WardenGnaw

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -514,17 +514,17 @@
                   "items": {
                     "type": "string"
                   },
-                  "description": "Array of symbol server URLs (example: http\u200b://MyExampleSymbolServer) or directories (example: /build/symbols) to search for .pdb files. These directories will be searched in addition to the default locations -- next to the module and the path where the pdb was originally dropped to.",
```
Comment (WardenGnaw) [inline package.json:517]: Undo removing `\u200b`

We use '\u200b' (unicode zero-length space character) to break VS Code's URL detection regex for URLs that are examples.

File: src/coreclr-debug/activate.ts [hunk 2]
```diff
@@ -28,6 +28,9 @@ export async function activate(thisExtension: vscode.Extension<CSharpExtensionEx
     } else if (!CoreClrDebugUtil.existsSync(_debugUtil.installCompleteFilePath())) {
         completeDebuggerInstall(platformInformation, eventStream);
     }
+
+    const factory = new DebugAdapterExecutableFactory(platformInformation, eventStream, thisExtension.packageJSON, thisExtension.extensionPath);
+    context.subscriptions.push(vscode.debug.registerDebugAdapterDescriptorFactory('coreclr', factory));
```
Comment (WardenGnaw) [inline src/coreclr-debug/activate.ts:33]: We will also need a DebugAdapterDescriptor for DesktopCLR debugging.

`context.subscriptions.push(vscode.debug.registerDebugAdapterDescriptorFactory('clr', factory));`

File: src/coreclr-debug/activate.ts [hunk 3]
```diff
@@ -104,49 +107,65 @@ function showDotnetToolsWarning(message: string): void {
...
+export class DebugAdapterExecutableFactory implements vscode.DebugAdapterDescriptorFactory {
 
-        if (!installLock) {
-            eventStream.post(new DebuggerNotInstalledFailure());
-            throw new Error('The C# extension is still downloading packages. Please see progress in the output window below.');
-        }
-        // install.complete does not exist, check dotnetCLI to see if we can complete.
-        else if (!CoreClrDebugUtil.existsSync(util.installCompleteFilePath())) {
-            let success: boolean = await completeDebuggerInstall(platformInfo, eventStream);
+    constructor(private readonly platformInfo: PlatformInformation, private readonly eventStream: EventStream, private readonly packageJSON: any, private readonly extensionPath: string) {
```
Comment (WardenGnaw) [inline src/coreclr-debug/activate.ts:115]: I'm not sure if this is a new Typescript feature, but are these arguments saved to the object?

Does `this.packageJSON`/`this.platformInfo` exist?

File: src/coreclr-debug/activate.ts (hunk 2 above)
Comment (JoeRobich) [reply src/coreclr-debug/activate.ts:33]: Doh, thanks!

File: src/coreclr-debug/activate.ts (hunk 3 above)
Comment (JoeRobich) [reply src/coreclr-debug/activate.ts:115]: It is a TypeScript shorthand. They are private fields.

Comment (WardenGnaw) [reply src/coreclr-debug/activate.ts:115]: This is amazing! I've always assigned parameters to the object. Now to never have to write `self.parameter = parameter` in Typescript again.

File: package.json (hunk 1 above)
Comment (JoeRobich) [reply package.json:517]: I'll fix it by running the update dependencies task.

File: src/coreclr-debug/activate.ts (hunk 2 above)
Comment (JoeRobich) [reply src/coreclr-debug/activate.ts:33]: Added registration for 'clr'

File: package.json (hunk 1 above)
Comment (JoeRobich) [reply package.json]: Fixed

File: src/coreclr-debug/activate.ts [hunk 4]
```diff
@@ -104,49 +108,65 @@ function showDotnetToolsWarning(message: string): void {
...
-    // debugger has finished installation, kick off our debugger process
-    return {
-        command: path.join(common.getExtensionPath(), ".debugger", "vsdbg-ui" + CoreClrDebugUtil.getPlatformExeExtension())
-    };
+        // debugger has finished installation, kick off our debugger process
+
+        // use the executable specified in the package.json if it exists or determine it based on some other information (e.g. the session)
+        if (!executable) {
+            const command = path.join(common.getExtensionPath(), ".debugger", "vsdbg-ui" + CoreClrDebugUtil.getPlatformExeExtension());
+            //const args = [
```
Comment (WardenGnaw) [inline src/coreclr-debug/activate.ts:158]: nit: leftover comments?

File: src/features/commands.ts [hunk 5]
```diff
@@ -52,8 +51,6 @@ export default function registerCommands(context: vscode.ExtensionContext, serve
     disposable.add(vscode.commands.registerCommand('csharp.listRemoteProcess', async (args) => RemoteAttachPicker.ShowAttachEntries(args, platformInfo)));
 
     // Register command for adapter executable command.
```
Comment (WardenGnaw) [inline src/features/commands.ts:53]: nit: remove outdated comment

**Pull request merged**